e2e/e2e_test.go

```diff
@@ -198,6 +198,28 @@ func TestGetAndPutE2E(t *testing.T) {
 			expectedCommitCount: 9,
 			expectedCommits:     []string{"Push 2."},
 		},
+		{
+			description: "get works when checkout",
+			source: resource.Source{
+				Repository:  "itsdalmo/test-repository",
+				V3Endpoint:  "https://api.github.com/",
+				V4Endpoint:  "https://api.github.com/graphql",
+				AccessToken: os.Getenv("GITHUB_ACCESS_TOKEN"),
+			},
+			version: resource.Version{
+				PR:            targetPullRequestID,
+				Commit:        targetCommitID,
+				CommittedDate: time.Time{},
+			},
+			getParameters: resource.GetParameters{
+				IntegrationTool: "checkout",
+			},
+			putParameters:       resource.PutParameters{},
+			versionString:       `{"pr":"4","commit":"a5114f6ab89f4b736655642a11e8d15ce363d882","committed":"0001-01-01T00:00:00Z"}`,
+			metadataString:      `[{"name":"pr","value":"4"},{"name":"url","value":"https://github.com/itsdalmo/test-repository/pull/4"},{"name":"head_name","value":"my_second_pull"},{"name":"head_sha","value":"a5114f6ab89f4b736655642a11e8d15ce363d882"},{"name":"base_name","value":"master"},{"name":"base_sha","value":"93eeeedb8a16e6662062d1eca5655108977cc59a"},{"name":"message","value":"Push 2."},{"name":"author","value":"itsdalmo"}]`,
+			expectedCommitCount: 7,
```

Comment:
@itsdalmo so I've created a e2e test case for `checkout` but i'm kind of unclear where this `expectedCommitCount` is coming from.
it's definitely different from `merge` and `rebase`, and I've just let the test guided me to put 7 but it would be nice to know what this number means.

Reply:
If you check the code it looks like is specific of this repo. And is counting all the commits, branches and tags.

Reply:
Hi! The `expectedCommitCount` is the expected number of commits in the git history after a `get`. Instead of writing out the complete expected git history, I decided to just check the number of commits (`expectedCommitCount`), and specific commits in `expectedCommits`. 

Perhaps it would be best to just write out the entire Git history for all the test cases, it's just ~7-9 lines for each case 🤔

Reply:
But they will be repeated. 🤔

Reply:
okay, got it. that was what i was suspecting when searching the commits so thanks for clarifying.

i think this PR should be ready now so could you take another look at it again?